python/ray/_private/structured_logging.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import ray
+import json
+
+from ray._private.ray_constants import (
+    LOG_RAY_CORE_CONTEXT,
+    LOGGER_RAY_ATTRS,
+    LOGRECORD_STANDARD_ATTRS,
+)
+
+
+class CoreContextFilter(logging.Filter):
+    def filter(self, record):
+        runtime_context = ray.get_runtime_context()
+        ray_core_context = {
+            "job_id": runtime_context.get_job_id(),
+            "worker_id": runtime_context.get_worker_id(),
+            "node_id": runtime_context.get_node_id(),
+        }
+        if runtime_context.worker.mode == ray.WORKER_MODE:
+            ray_core_context["actor_id"] = runtime_context.get_actor_id()
+            ray_core_context["task_id"] = runtime_context.get_task_id()
+        setattr(record, LOG_RAY_CORE_CONTEXT, ray_core_context)
```

Comment:
We can directly set those attributes to record itself, no need an extra layer of LOG_RAY_CORE_CONTEXT.

Reply:
See here for more details https://github.com/ray-project/ray/pull/45098#discussion_r1587052052. Using `record.ray_core` will make it hard to maintain.

Reply:
Oh, I mean just `setattr(record, "actor_id", ....)`

Reply:
Oh, I misunderstood your point. I use an extra layer of dictionaries because this makes the Ray AI libraries easier to use with our formatter. The formatter needs to determine which fields are Ray-specific, which are user-defined, and which are LogRecord's defaults. 

If we expand all Ray-specific fields directly into `record`, we will need to maintain a large `set` of all Ray-specific attributes. Then, users also need to update the `set` whenever they add or delete an attribute. With the extra layer, users only need to update `LOGGER_RAY_ATTRS` once.

For example, Ray Serve can add `ray_serve` into `LOGGER_RAY_ATTRS`, and then they just need to maintain their `ray_serve` dictionary in the LogRecord. If we don't have the dictionary and Ray Serve wants to add a new attribute `request_id` to the LogRecord, they need to (1) update the `set` for Ray-specific attributes (2) update LogRecord.

If maintaining a large set is OK, I can update the PR.

Reply:
The issue is that the output json dictionary is flatten so we need to make sure all the keys are unique and don't conflict with each other unless the output json is also `{"ts": "xxx", "ray_core": {}, "ray_serve": {}}`

Reply:
I will update it.

Reply:
Update https://github.com/ray-project/ray/pull/45098/commits/43bfc5aca840d998365ee25d9cbd5b1fe47f68b5